src/modules/RA/DataLayer/components/tabs/StyleTab/Style/ColorListField.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import randomColor from 'randomcolor';
+
+import { makeStyles } from '@material-ui/core/styles';
+
+import ColorPicker from '../../../../../../../components/react-admin/ColorPicker';
+
+const useStyles = makeStyles({
+  colorList: {
+    display: 'flex',
+    '& > *': {
+      margin: '5px',
+    },
+    '& .action': {
+      width: '25px',
+      height: '25px',
+      backgroundColor: 'none',
+      border: '1px dotted #ccc',
+      padding: '0px',
+    },
+  },
+});
+
+const ColorListField = ({ value, onChange }) => {
+  const classes = useStyles();
+
+  const handleColorChange = index => newValue => {
+    const newColorList = [...value];
+    newColorList[index] = newValue;
+    onChange(newColorList);
+  };
+
+  const addColor = () => {
+    const newColorList = [...value];
+    newColorList.push(randomColor());
+    onChange(newColorList);
+  };
+
+
+  const removeColor = index => () => {
+    const newColorList = [...value];
+    newColorList.splice(index, 1);
+    onChange(newColorList);
+  };
```

Comment:
Maybe you could wrap each handler below by an useCallbak

Reply:
Not possible here as we have function that return function.